tests/FakeItEasy.Tests/Configuration/RuleBuilderTests.cs

```diff
@@ -52,7 +52,7 @@ public void Returns_called_with_value_returns_parent_configuration()
 
             var result = returnConfig.Returns(10);
 
-            result.Should().Be(this.builder);
+            result.Should().BeSameAs(returnConfig);

```

Comment:
No, we're returning the `ReturnValueConfiguration<T>` now (cf. the "other part of the surprise" in your comment above). Anyway, I think this test falls in the Gulliver's Travels Test category... from an external point of view, it shouldn't matter which object we return, as long as it has the correct interface.